cdap-tms/src/main/java/co/cask/cdap/messaging/store/RawMessageTableEntry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.messaging.store;
+
+import javax.annotation.Nullable;
+
+/**
+ * Container class that contains raw bytes corresponding to an entry in the Message Table.
+ */
+public class RawMessageTableEntry {
+  private byte[] key;
+  private byte[] txPtr;
+  private byte[] payload;
+
+  public RawMessageTableEntry clear() {
+    this.key = null;
+    this.txPtr = null;
+    this.payload = null;
+    return this;
+  }
+
+  public RawMessageTableEntry setKey(byte[] key) {
```

Comment:
Seems like we always call `clear()` followed by the three `set` methods. Probably cleaner if we have a single `RawMessageTableEntry set(byte[] key, @Nullable byte[] payload, @Nullable byte[] txPtr)` method.